lib/katello/tasks/upgrades/3.11/import_yum_metadata_file.rake

```diff
@@ -0,0 +1,16 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.11' do
+      desc "Import yum metadata file units from pulp"
+      task :import_yum_metadata => ["environment"] do
+        User.current = User.anonymous_api_admin
+        puts _("Importing Yum Metadata Files")
+        Katello::YumMetadataFile.import_all
+        Katello::Repository.linked_repositories.each do |repo|
+          base_repo = repo.target_repository
+          Katello::YumMetadataFile.copy_repository_associations(base_repo, repo)
```

Comment:
I believe if the user runs this on an already broken installation it will cause katello to continue not republishing the repository when it should.  What do you think about NOT doing this for linked_repositories?  This would force the check_matching_content check to fail for all lifecycle env repos on the next publish/promote, but would ensure that everything is cleaned up on the next promote.